exchange.go

```diff
@@ -442,7 +475,144 @@ func (re *RabbitExchangeImpl) Receive(exchange ExchangeSettings, queue QueueSett
 						closer()
 						for {
 							time.Sleep(time.Millisecond * 250)
-							channel, msgs, errChan, closer, err = getChannel()
+							channel, msgs, errChan, closer, err = re.getChannel(exchange, queue, retryExchangeName)
+							if err != nil {
+								log.Printf("Error connecting to rabbit %+v\n", err)
+							} else {
+								break
+							}
+						}
+					}
+
+				}
+			}
+		},
+		func() {
+			close(stop)
+		}, nil
+}
+
+func (re *RabbitExchangeImpl) BulkReceive(exchange ExchangeSettings, queue QueueSettings, maxWait int) (func(BulkMessageHandleFunc) error, func(), error) {
+	retryExchangeName := fmt.Sprintf("%s-%s", exchange.Name, retryExchangeNameSuffix)
+
+	channel, msgs, errChan, closer, err := re.getChannel(exchange, queue, retryExchangeName)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	stop := make(chan struct{})
+	return func(handler BulkMessageHandleFunc) error {
+			defer closer()
+			batch := make([]amqp.Delivery, 0, queue.Prefetch)
+			timerDuration := time.Duration(maxWait) * time.Millisecond
+			fillWaitTimer := time.NewTimer(timerDuration)
```

Comment:
```suggestion
			fillWaitTimer := time.NewTimer(maxWait)
```

Now that `maxWait` is a time.Duration, we can just use that here.

Reply:
And ideally we would inject the `NewTimer` function into the service, so we can mock time in the tests. 
Not needed for this PR.

Reply:
Will add all suggestions that don't need to be added in this PR as TODOs